### ISSUE TITLE
Support option for getting arrows at the end of rules

### ISSUE THREAD

**Issue by davidanthoff** [vega/vega-lite#4270]
Then one could use rule marks to create nice quiver plots. I think it would really be two styling options: one for the beginning, and one for the end of the rule, and then a selection of arrows or other symbols.

**Comment by domoritz** [comment]
Yeah, that would be nice. I can imagine that we support symbols at the beginning and end of rule marks. I can imagine that this is a property of the rule mark. 

We don't have the bandwidth to add this right now but would be happy to guide someone.

**Comment by davidanthoff** [comment]
Just FYI, this came up [here](https://github.com/fredo-dedup/VegaLite.jl/issues/130). In that case one can actually achieve what the original question hoped to do with the shipping version, but I can also see that people will want proper quiver plots. No rush :) Wrap faceting is what I really have my fingers crossed for ;)

**Comment by kanitw** [comment]
Note that this will require support for angle for symbols in Vega for
non-vertical/horizontal rules.

**Comment by domoritz** [comment]
The simplest way is probably to just add support for symbol angles and then you could use layer to add the arrow ends. Angle would be a new channel that only works with point marks (and only make sense for symbols). 

> Wrap faceting is what I really have my fingers crossed for ;)

[Me too!](https://github.com/vega/vega-lite/issues/393)

**Comment by davidanthoff** [comment]
I can see the two layer approach working, but on the other hand, it would be much easier for users if this was just a property of rules. Also, I'm just wondering whether something that is stitched together might have some visual "breaks". Things like, does the rule end exactly within the arrow, or is there some overlap/space might be quite tricky if it is just a layered story. Also, would be nice if things like line width would flow to the arrow as well...

**Comment by lovettchris** [comment]
I would love to see "angle' get added for symbols too, I can get so close to what I need with a triangle mark in a second layer, just need to be able to rotate it...

![image](https://user-images.githubusercontent.com/18707114/56088534-ed5cb980-5e37-11e9-93d2-77e85f3a6002.png)

**Comment by pierre-haessig** [comment]
For the reference, an arrow annotation example has been merged in Altair doc by https://github.com/altair-viz/altair/pull/3236: see [Examples/Line Chart with Arrows](https://altair-viz.github.io/gallery/line_chart_with_arrows.html).

However, about the previous comments about the possibility of using mark angles (which have been implemented in the mean time, see e.g. [Wind Vector Map example](https://vega.github.io/vega-lite/examples/point_angle_windvector.html)), there is a catch. Indeed, as pointed out by @ChiaLingWeng (https://github.com/altair-viz/altair/pull/3236#issuecomment-1782355793), the *arrow angles in a vector plot depend on the aspect ratio* of that plot. I guess this is difficult to implement with the mark angles.

So in the end, having the option to add arrow heads at one or both ends of rule marks would still be a useful contribution.
